src/main/java/org/cyclops/integratedterminals/client/gui/container/ContainerScreenTerminalStorage.java

```diff
@@ -846,6 +847,29 @@ public int getStorageSlotIndexAtPosition(double mouseX, double mouseY) {
         return -1;
     }
 
+    /**
+     * Returns the rectangle that a storage slot occupies even if the slot is not visible.
+     * Use {@link ContainerScreenTerminalStorage#getStorageSlotIndexAtPosition(double, double)} to get a currently visible slot.
+     * @param slotIndex
+     * @return {@link Rect2i} of the slot.
+     */
+    public Rect2i getStorageSlotRect(int slotIndex) {
```

Comment:
Just wondering; is this method really needed here?
Would there be a downside to just moving this one to the JEI compat?

Reply:
I decided to put it here because most of the GUI helper functions I used are `protected`. I'd duplicate the `protected` logic in the compat mod if that's more convenient.

Reply:
Ah, didn't know they were protected. We could also make them public, but I'm fine with the approach currently taken.
Will look into it in more detail later!